programs/uxd/src/instructions/mint.rs

```diff
@@ -203,24 +205,28 @@ struct DepositoryInfoForMint<'info> {
 
 pub(crate) fn handler(ctx: Context<Mint>, collateral_amount: u64) -> Result<()> {
     // Gather all the onchain states we need (caps, weights and supplies)
-    let controller = ctx.accounts.controller.load()?;
+    let mut controller = ctx.accounts.controller.load_mut()?;
     let identity_depository = ctx.accounts.identity_depository.load()?;
     let mercurial_vault_depository = ctx.accounts.mercurial_vault_depository.load()?;
     let credix_lp_depository = ctx.accounts.credix_lp_depository.load()?;
 
+    // Make sure minting reduces the counter of outflow (minting is inflow)
+    controller.epoch_outflow_amount = if controller.epoch_outflow_amount > collateral_amount {
```

Comment:
Maybe this is more explicit:

```suggestion
    controller.epoch_outflow_amount = min(0, checked_sub(controller.epoch_outflow_amount, collateral_amount)?);
```

Reply:
Sorry I don't know how to use suggestion properly

Reply:
We could make a math function for this :D

Reply:
Ok no math function we'r good I take it back x)